Code/Tools/AssetProcessor/native/unittests/AssetScannerUnitTests.cpp

```diff
@@ -7,130 +7,145 @@
+        AssetScanner scanner(&config);
+
+        QList<AssetFileInfo> filesFound;
+        QList<AssetFileInfo> foldersFound;
+
+        bool doneScan = false;
+
+        connect(
+            &scanner, &AssetScanner::FilesFound, this,
+            [&filesFound](QSet<AssetFileInfo> fileList)
+            {
+                for (AssetFileInfo foundFile : fileList)
+                {
+                    filesFound.push_back(foundFile);
+                }
+            });
+
+        connect(
+            &scanner, &AssetScanner::FoldersFound, this,
+            [&foldersFound](QSet<AssetFileInfo> folderList)
+            {
+                for (AssetFileInfo foundFolder : folderList)
+                {
+                    foldersFound.push_back(foundFolder);
+                }
+            });
+
+        connect(
+            &scanner, &AssetScanner::AssetScanningStatusChanged, this,
+            [&doneScan](AssetProcessor::AssetScanningStatus status)
+            {
+                if ((status == AssetProcessor::AssetScanningStatus::Completed) || (status == AssetProcessor::AssetScanningStatus::Stopped))
+                {
+                    doneScan = true;
+                }
+            });
+
+        // this test makes sure that no files that should be missed were actually missed.
+        // it makes sure that if a folder is added recursively, child files and folders are found
+        // it makes sure that if a folder is added NON-recursively, child folder files are not found.
+
+        scanner.StartScan();
+        QElapsedTimer nowTime;
+        nowTime.start();
+        while (!doneScan)
         {
-            foldersFound.push_back(foundFolder);
-        }
-    }
-    );
+            QCoreApplication::processEvents(QEventLoop::WaitForMoreEvents, 100);
 
-    connect(&scanner, &AssetScanner::AssetScanningStatusChanged, this, [&doneScan](AssetProcessor::AssetScanningStatus status)
-        {
-            if ((status == AssetProcessor::AssetScanningStatus::Completed) || (status == AssetProcessor::AssetScanningStatus::Stopped))
+            if (nowTime.elapsed() > 10000)
             {
-                doneScan = true;
+                break;
             }
         }
-        );
 
-    // this test makes sure that no files that should be missed were actually missed.
-    // it makes sure that if a folder is added recursively, child files and folders are found
-    // it makes sure that if a folder is added NON-recursively, child folder files are not found.
+        EXPECT_TRUE(doneScan);
+        EXPECT_TRUE(filesFound.count() == expectedFiles.count());
 
-    scanner.StartScan();
-    QElapsedTimer nowTime;
-    nowTime.start();
-    while (!doneScan)
-    {
-        QCoreApplication::processEvents(QEventLoop::WaitForMoreEvents, 100);
-
-        if (nowTime.elapsed() > 10000)
+        for (const AssetFileInfo& file : filesFound)
```

Comment:
As above, something like this can probably be converted to something like EXPECT_THAT(filesFound, Contains(file.m_filePath)) which would give a much nicer error message.  I'm fine if these are left for later though since they're not as straightforward as the EXPECT_EQ type macros

Reply:
amzn-mike and I talked about this. EXPECT_THAT won't work in this case, the containers hold different types. I hadn't heard of EXPECT_THAT and ended up learning a ton about it, it seems great for the times you can use it.

Reply:
EXPECT_THAT can be used to compare containers of different types.
See [SettingsRegistryMergeUtilsTests.cpp](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/Tests/Settings/SettingsRegistryMergeUtilsTests.cpp#L542-L543) for an example of comparing a `fixed_vector<const char*, 8>` vs an `AZStd::vector<AZ:IO::Path>`